File: source_funcs/go_transcoder/LARGEST_SUM_CONTIGUOUS_SUBARRAY_2/LARGEST_SUM_CONTIGUOUS_SUBARRAY_2_emit.go
package main

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

//go:export f_gold
func f_gold(a []int, size int) int {
	var (
		max_so_far int = a[0]
		curr_max   int = a[0]
	)
	for i := int(1); i < size; i++ {
		curr_max = max(a[i], curr_max+a[i])
		max_so_far = max(max_so_far, curr_max)
	}
	return max_so_far
}

File: rust_funcs/c_transcoder/LARGEST_SUM_CONTIGUOUS_SUBARRAY_2/LARGEST_SUM_CONTIGUOUS_SUBARRAY_2_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


#[no_mangle]
fn f_gold(a: [i32;2], size: i32) -> i32 {
   let mut max_so_far = a[0 as usize];
   let mut curr_max = a[0 as usize];
   for i in 0..size as usize{
       curr_max = max(a[i], curr_max + a[i] as i32);
       max_so_far = max(max_so_far, curr_max);
   }
   max_so_far 
}
